### PR TITLE
Add basic message type for AMQP

### PR DESCRIPTION
**Pull request by deprov447**
This will (eventually) add support for AMQP protocol parsing to pixie.

Signed-off-by: Anubhav Choudhary <ac.10edu@gmail.com>

### CODE REVIEW

File: src/stirling/source_connectors/socket_tracer/protocols/amqp/types.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+namespace px {
+namespace stirling {
+namespace protocols {
+namespace amqp {
+
+// Represents a generic AMQP message.
+struct Message : public FrameBase {
+  // Defines the type of message passed [takes values 1...4]
+  enum type { Method = 1, Header = 2, Body = 3, Heartbeat = 4 };
```
Comment (yzhao1012) [inline src/stirling/source_connectors/socket_tracer/protocols/amqp/types.h:34]: nit: Use `enum class Type { kMethod = 1, kHeader = 2, kBody = 3, kHeartbeat = 4 };`

This is just the existing style in the codebase.

File: src/stirling/source_connectors/socket_tracer/protocols/amqp/types.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  // Communication channel to be used
+  short int channel;
```
Comment (yzhao1012) [inline src/stirling/source_connectors/socket_tracer/protocols/amqp/types.h:37]: nit: Use fixed size integer type, for example `int16_t`, this is another existing style in the codebase, where we use fixed size type when it's describing the protocol's data obtained from inside kernel.

File: src/stirling/source_connectors/socket_tracer/protocols/amqp/types.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  // Defines the length of message upcoming
+  long int message_length;
```
Comment (yzhao1012) [inline src/stirling/source_connectors/socket_tracer/protocols/amqp/types.h:40]: nit: Use int32_t

File: src/stirling/source_connectors/socket_tracer/protocols/amqp/types.h [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+  // Actual body content to be used
+  std::string_view message_body;
+
+  // Marks end of the frame by hexadecimal value %xCE
+  const std::string Frame_end{"CE"};
```
Comment (yzhao1012) [inline src/stirling/source_connectors/socket_tracer/protocols/amqp/types.h:46]: nit: Use the following:
```
static constexpr std::string_view kFrameEnd = "CE"
```

**Comment by yzhao1012** [comment]
Please rename the PR title to
```
Add basic message type for AMQP
```

Otherwise, it looks good.

**Review by yzhao1012 — APPROVED** [review]
The only thing left is the PR title, if it could be changed to match the actual change, we can land it.

**Comment by deprov447** [comment]
done

**Review by yzhao1012 — APPROVED** [review]

**Comment by vihangm** [comment]
Closed by d166576e3533e3ec85e361027791a485ea810c4a

**Pull request closed without merging**